src/vs/base/common/lifecycle.ts

```diff
@@ -173,6 +198,13 @@ export function toDisposable(fn: () => void): IDisposable {
 	return self;
 }
 
+/**
+ * Manages a collection of disposable values.
+ *
+ * This is the preferred way to manage multiple disposables. A `DisposableStore` is safer to work with than a
+ * `IDisposable[]` as it considers edge cases, such as registering the same value multiple times or what happens
+ * if you try adding an item to a store that has already been disposed of.
```

Comment:
nit: "such as registering the same value multiple times or adding an item to a store that has already been disposed of"